gnucash/gnucash-commands.cpp

```diff
@@ -77,22 +78,30 @@ scm_add_quotes(void *data, [[maybe_unused]] int argc, [[maybe_unused]] char **ar
 {
     auto add_quotes_file = static_cast<const std::string*>(data);
 
-    scm_c_eval_string("(debug-set! stack 200000)");
-
-    auto mod = scm_c_resolve_module("gnucash price-quotes");
-    scm_set_current_module(mod);
-
     gnc_prefs_init ();
     qof_event_suspend();
-    scm_c_eval_string("(gnc:price-quotes-install-sources)");
 
-    if (!gnc_quote_source_fq_installed())
+    auto quotes = gnc_get_quotes_instance();
+    if (quotes.cmd_result() == 0)
+    {
+        std::cout << bl::format (bl::translate ("Found Finance::Quote version {1}.")) % quotes.version() << std::endl;
+        auto quote_sources = quotes.sources_as_glist();
+        gnc_quote_source_set_fq_installed (quotes.version().c_str(), quote_sources);
+        g_list_free (quote_sources);
```

Comment:
```suggestion
        g_list_free_full (quote_sources, g_free);
```

Reply:
Ah,  good catch!
I initially intended to work directly with the  const char* strings returned  from a std::string via  c_str(), but those can't be added in a GList (as  GList data is not  const). So when I rewrote the for_each function I added a g_strdup and  indeed that requires all data to be freed.

This will become a non-issue when I rewrite gnc-commodity to use the GncQuotes object directly, but for now let's keep things tidy.

Reply:
Ah, good catch!
My initial implementation used const  char* strings, but with the for_each replacement I had  to switch to  duplicated char* strings because GList discards const information. So  yes it now needs g_list_free_full to avoid extra memory leaks.

This will no longer be relevant once gnc-commodity interfaces directly with GncQuotes via C++ methods. But until then, let's keep it all tidy  :)

Reply:
Ok, I replied to this from two separate browser windows... Thought my initial  reply had vanished.